### PR TITLE
kotlin2cpg: add support for anonymous functions

### PR DESCRIPTION
**Pull request by ursachec**
these are different from lambdas https://kotlinlang.org/docs/lambdas.html#anonymous-functions

### CODE REVIEW

File: joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/AnonymousFunctionsTests.scala [hunk 1]
```diff
@@ -1,31 +1,82 @@
...
 
 import io.joern.kotlin2cpg.testfixtures.KotlinCode2CpgFixture
+import io.shiftleft.codepropertygraph.generated.{EvaluationStrategies, ModifierTypes}
 import io.shiftleft.semanticcpg.language._
 import org.scalatest.Ignore
 
 import scala.annotation.unused
 
-@Ignore
+//@Ignore
```
Comment (xavierpinho) [inline joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/AnonymousFunctionsTests.scala:10]: Delete comment?

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/KtPsiToAst.scala [hunk 2]
```diff
@@ -561,6 +561,91 @@ trait KtPsiToAst {
...
+    case class NodeContext(node: NewNode, name: String, typeFullName: String)
+    val closureBindingEntriesForCaptured = scope
+      .pushClosureScope(lambdaMethodNode)
+      .collect {
+        case node: NewMethodParameterIn => NodeContext(node, node.name, node.typeFullName)
+        case node: NewLocal             => NodeContext(node, node.name, node.typeFullName)
+        case node: NewMember            => NodeContext(node, node.name, node.typeFullName)
+      }
+      .map { capturedNodeContext =>
+        // TODO: remove the randomness here, two CPGs created from the same codebase should be the same
```
Comment (xavierpinho) [inline joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/KtPsiToAst.scala:579]: TODO for a future PR, right?

**Review by xavierpinho — APPROVED** [review]

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/KtPsiToAst.scala (hunk 2 above)
Comment (ursachec) [reply joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/KtPsiToAst.scala:579]: yup

File: joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/AnonymousFunctionsTests.scala (hunk 1 above)
Comment (ursachec) [reply joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/AnonymousFunctionsTests.scala:10]: yes, thx

**Pull request merged**